src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -73,8 +70,8 @@ internal class FhirCosmosSearchService : SearchService
             _physicalPartitionInfo = physicalPartitionInfo;
             _queryPartitionStatisticsCache = queryPartitionStatisticsCache;
             _logger = logger;
-            _resourceTypeSearchParameter = searchParameterDefinitionManager.GetSearchParameter(KnownResourceTypes.Resource, SearchParameterNames.ResourceType);
-            _resourceIdSearchParameter = searchParameterDefinitionManager.GetSearchParameter(KnownResourceTypes.Resource, SearchParameterNames.Id);
+            _resourceTypeSearchParameter = SearchParameterInfo.ResourceTypeSearchParameter;
+            _resourceIdSearchParameter = new SearchParameterInfo(SearchParameterNames.Id, SearchParameterNames.Id);
```

Comment:
>resourceIdSearchParameter = new SearchParameterInfo(SearchParameterNames.Id, SearchParameterNames.Id); [](http://example.com/codeflow?start=13&length=102)

Are we sure that downstream code that uses this (since we are passing it as part of an Expression object) will only need the `Name` and `Code` property of the `SearchParamInfo` object?

Reply:
Well, we have plenty of tests going through this part of code, so I somewhat confident.